dali/base/dadfs.cpp

```diff
@@ -5792,15 +5792,23 @@ class CDistributedSuperFile: public CDistributedFileBase<IDistributedSuperFile>
     __int64 getRecordCount()
     {
         __int64 ret = queryAttributes().getPropInt64("@recordCount",-1);
-        if (ret==-1) {
+        if (ret==-1)
+        {
             ret = 0;
-            ForEachItemIn(i,subfiles) {
+            ForEachItemIn(i,subfiles)
+            {
+                IDistributedFile &subFile = subfiles.item(i);
                 __int64 rc = subfiles.item(i).queryAttributes().getPropInt64("@recordCount",-1);
-                if (rc == -1) {
-                    ret = rc;
-                    break;
+                if (rc == -1)
+                {
+                    IDistributedSuperFile *super = subFile.querySuperFile();
+
+                    // if regular file or non-empty super, must have a @recordCount to aggregate a total
+                    if ((nullptr == super) || (super->numSubFiles()>0))
```

Comment:
if super->numSubFiles()>0 should we walk that and aggregate its recordCount ?

Reply:
'super' should have an aggregate record count - if it does not, it means that it failed to aggregate for some similar reason, e.g. because one of it's subs had a missing recordCount.
So can only continue here (where no recordCount attribute) if it's an empty super.